lizmap/modules/admin_api/lib/RepoCreator.php

```diff
@@ -41,43 +47,63 @@ public static function createRepository(string $key, string $label, string $path
         // Testing a relative path and updating it if needed
         try {
             $path = self::testRelativePath($path, $rootRepositories);
+        } catch (ApiException $e) {
+            throw new ApiException($e->getMessage(), $e->getCode());
         } catch (\Exception $e) {
             throw new \Exception($e->getMessage(), $e->getCode());
         }
 
-        if (!file_exists($path) or !is_dir($path)) {
-            throw new \Exception(
-                "The path provided doesn't exist or is not a directory ! ".
-                "It has to be a folder in '{$rootRepositories}'.",
-                400
+        if (!is_dir($path)) {
+            if ($createDirectory) {
+                $oldUmask = umask(0002);
+                $isCreated = mkdir($path, 0775, true);
+                umask($oldUmask);
+                if (!$isCreated) {
+                    throw new ApiException(
+                        "Unable to create directory '{$path}'".
+                        " Maybe you don't have enough permissions.",
```

Comment:
```suggestion
                        " Maybe the process running Lizmap does not have enough permissions.",
```
I think it's not the connected user, but the process, usually `www-data` ?

And maybe a message should be send to jlog `error` : "The user "foo" tried to create the directory "/bar/..." but got the error ...." ?

Every-time an `Exception` is catch because this error were not expected, nice to keep some log about it.